### ISSUE TITLE
Traversing materials, help needed

### ISSUE THREAD

**Issue by Chrjstoph** [ingowald/pbrt-parser#4]
Hi,

I want to traverse the Materials and extract a texture.

auto kd = pbrtMat->findParam < ??? >("Kd");

I am struggling to identify the proper type and vs2013 is silent. 
kd is always empty.
Stepping through the source shows that the data is available prior to the dynamic cast.

**Comment by Chrjstoph** [comment]
Never mind. 
I did not account for the other materials that actually require a float and come prior to the texture material.

ParamT<std::string> appears to be the way to parse the texture related stuff.

**Comment by Chrjstoph** [comment]
On another note: Do you parse Textures yet? And if so, how are they accessed.

As I see it, the Parser class does not yet give access to the Textures and namedMaterials.
I assume you will want to give the pointers similar to getScene().

**Comment by ingowald** [comment]
I actually do have a 'devel' branch as well; in this branch I recently fleshed out the 'pbrt2rivl' exporter to export both materials and textures. There's a 'picopng' parser in there that reads the textures, plus all the code to compute the path, load the data, etc. 

I haven't done too much testing with it, though - this is a side project I do for fun, so I have limited time - and ospray cannot actualy load those generated rivl files, yet (becuase it doesn't support the pbrt materials, yet). From eyeballing it it looks like the textures _are_ handles properly, but again: i haven't tested it yet.

Also, this branch isn't run through the windows-grinder yet, so it may well not compile under vs.
